tsuru/client/deploy.go

```diff
@@ -290,13 +366,36 @@ func targz(ctx *cmd.Context, destination io.Writer, filepaths ...string) error {
 		if err != nil {
 			return err
 		}
+		wd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		fiName := filepath.Join(wd, fi.Name())
 		if fi.IsDir() {
+			for _, p := range ignoreList {
```

Comment:
It would be more efficient for `ignoreList` to be a `var ignoreSet map[string]struct{}`. This way you could ignore patterns in O(1) by doing:
```
if _, inSet := ignoreSet[fiName]; !inSet {
    continue
}
```